### PR TITLE
xplat: runtests.py improvements

### PR DESCRIPTION
**Pull request by jianchun**
1. Add error handling

 Python multiprocessing parallel processes do not print anything if an
 unhandled exception occurs. This causes the script to "hang" silently.
 Added 2 levels of try/except so that the script no longer hangs on
 exception. Script exceptions and backtraces will be reported and logged.

2. Switch to python3 `bytes` for output/baseline diff

 This has no effect on python 2, but enables the script on python 3.
 Capture output as bytes (subprocess communicate() returned type), load
 baseline files in mode 'rb' and diff `bytes` type directly. This avoids
 issues caused by encoding/end-of-line conversions.

3. Refactor, notably move 'folder' data to an attribute of test so that
we don't need to pass folder around.

4. Add a tag "xplatslow" to 2 tests (temporarily, so can be excluded on
command line: `--not-tag xplatslow`).

### CODE REVIEW

File: test/runtests.py [hunk 1]
```diff
@@ -347,65 +381,60 @@ def timeout_func(timeout_data):
...
-                    baseline_output = bs_file.read()
-
-                # Cleanup carriage return
-                # todo: remove carriage return at the end of the line
-                #       or better fix ch to output same on all platforms
-                expected_output = re.sub('[\r]+\n', '\n', baseline_output)
-
-                # todo: implement wild cards support
+                with open(baseline, 'rb') as bs_file:
+                    expected_output = bs_file.read().replace(b'\r', b'')
```
Comment (obastemur) [inline test/runtests.py]: Old implementation only replaces CR(+)LF to LF, however the new one removes all the CR from the string. What would happen, if we had a test string with CR inside?

File: test/typedarray/rlexe.xml [hunk 2]
```diff
@@ -215,13 +215,13 @@ Below test fails with difference in space. Investigate the cause and re-enable t
   <test>
     <default>
       <files>allocation.js</files>
-      <tags>typedarray,exclude_arm</tags>
+      <tags>typedarray,exclude_arm,xplatslow</tags>
```
Comment (obastemur) [inline test/typedarray/rlexe.xml]: The reason these tests were slow in fact due to OOM fixed by https://github.com/Microsoft/ChakraCore/pull/1086 .At this point I don't see a reason to disable these.

**Comment by obastemur** [comment]
> ...This avoids issues caused by encoding/end-of-line conversions.

Can you please elaborate how this actually works? IMO, binary mode doesn't make a difference in terms of utf-16 BOM / endiannes. For end-of-line left a comment in place.

**Comment by jianchun** [comment]
> Can you please elaborate how this actually works? IMO, binary mode doesn't make a difference in terms of utf-16 BOM / endiannes. For end-of-line left a comment in place.

I mean diffing bytes rather than diffing decoded Unicode strings. I sent and closed a PR last week in which I `decode('utf-8')` and then diff the strings. That didn't work as some corner cases were not in utf-8.

**Comment by obastemur** [comment]
>> I mean diffing bytes rather than diffing decoded Unicode strings.

>> load baseline files in mode 'rb' and diff bytes type directly.

@jianchun , I must be missing that part somehow. Could you please point me out where actually you do that?

And eventually how it's going to help with saved baseline files ? Especially, considering the fact utf-16 BOM defines the order there. (let's forget about utf-8 BOM)

File: test/runtests.py (hunk 1 above)
Comment (jianchun) [inline test/runtests.py]: One reason I changed this is to be consistent with what we do when capturing ch output (`proc.communicate()[0].replace`). Testing CR with baseline is corner case / problematic and we should not do.

**Comment by jianchun** [comment]
> @jianchun , I must be missing that part somehow. Could you please point me out where actually you do that?

All the added `b`.

> And eventually how it's going to help with saved baseline files ? Especially, considering the fact utf-16 BOM defines the order there. (let's forget about utf-8 BOM)

It does not help with the problem of baseline files in different encodings. Existing ch output and baseline files are not in certain encoding. We could fix that in future and then we can decode here; but currently decoding (ch output) runs into errors.

File: test/typedarray/rlexe.xml (hunk 2 above)
Comment (jianchun) [inline test/typedarray/rlexe.xml]: That's great! I'll hold on this / revert after your fix.

**Comment by obastemur** [comment]
:shipit:

**Comment by jianchun** [comment]
@dotnet-bot test this please
